src/data_provider/src/packages/berkeleyRpmDbHelper.h

```diff
@@ -230,7 +231,13 @@ class BerkeleyRpmDBReader final
                         {
                             for (int i = 0; i < it->count; ++i)
                             {
-                                dirindexes.push_back(Utils::toInt32BE(reinterpret_cast<uint8_t*>(ucp)));
+                                int index = Utils::toInt32BE(reinterpret_cast<uint8_t*>(ucp));
+
+                                if (index >= 0 && index < DIRINDEX_MAX)
```

Comment:
Would it make sense to check this index against the input, rather than a hardcoded limit?